plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/bootstrap/FlightService.java

```diff
@@ -0,0 +1,170 @@
+ */
+public class FlightService extends NetworkPlugin.AuxTransport {
+    private static final Logger logger = LogManager.getLogger(FlightService.class);
+    private final ServerComponents serverComponents;
+    private StreamManager streamManager;
+    private Client client;
+    private FlightClientManager clientManager;
+    private SecureTransportSettingsProvider secureTransportSettingsProvider;
+    private BufferAllocator allocator;
+    private ThreadPool threadPool;
+
+    /**
+     * Constructor for FlightService.
+     * @param settings The settings for the FlightService.
+     */
+    public FlightService(Settings settings) {
+        Objects.requireNonNull(settings, "Settings cannot be null");
+        try {
+            ServerConfig.init(settings);
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to initialize Arrow Flight server", e);
+        }
+        this.serverComponents = new ServerComponents(settings);
+    }
+
+    void setClusterService(ClusterService clusterService) {
+        serverComponents.setClusterService(Objects.requireNonNull(clusterService, "ClusterService cannot be null"));
+    }
+
+    void setNetworkService(NetworkService networkService) {
+        serverComponents.setNetworkService(Objects.requireNonNull(networkService, "NetworkService cannot be null"));
+    }
+
+    void setThreadPool(ThreadPool threadPool) {
+        this.threadPool = Objects.requireNonNull(threadPool, "ThreadPool cannot be null");
+        serverComponents.setThreadPool(threadPool);
+    }
+
+    void setClient(Client client) {
+        this.client = client;
+    }
+
+    void setSecureTransportSettingsProvider(SecureTransportSettingsProvider secureTransportSettingsProvider) {
+        this.secureTransportSettingsProvider = secureTransportSettingsProvider;
+    }
+
+    /**
+     * Starts the FlightService by initializing the stream manager.
+     */
+    @Override
+    protected void doStart() {
+        try {
+            allocator = AccessController.doPrivileged((PrivilegedAction<BufferAllocator>) () -> new RootAllocator(Integer.MAX_VALUE));
+            serverComponents.setAllocator(allocator);
+            SslContextProvider sslContextProvider = ServerConfig.isSslEnabled()
+                ? new DefaultSslContextProvider(secureTransportSettingsProvider)
+                : new DisabledSslContextProvider();
+            serverComponents.setSslContextProvider(sslContextProvider);
+            serverComponents.initComponents();
+            clientManager = new FlightClientManager(
+                allocator, // sharing the same allocator between server and client
+                serverComponents.clusterService,
+                sslContextProvider,
+                serverComponents.workerEventLoopGroup, // sharing the same worker ELG between server and client
+                threadPool,
+                client
+            );
+            initializeStreamManager(clientManager);
```

Comment:
Why are we doing this?

Reply:
to initialize StreamManager with clientManager for it use them to handle any streaming request outside of local node.